picard/acoustid/__init__.py

```diff
@@ -211,7 +211,9 @@ def _on_fpcalc_finished(self, task, exit_code, exit_status):
         try:
             self._running -= 1
             self._run_next_task()
-            if exit_code == 0 and exit_status == QtCore.QProcess.ExitStatus.NormalExit:
+            # fpcalc returns the exit code 3 in case of decoding errors that
+            # still allowed it to calculate a result.
+            if exit_code in (0, 3) and exit_status == QtCore.QProcess.ExitStatus.NormalExit:
```

Comment:
```suggestion
            if exit_code in {0, 3} and exit_status == QtCore.QProcess.ExitStatus.NormalExit:
```

Instead of using numeric values we could define a bunch of int enum constants like `FPCALC_EXIT_NOERROR`, `FPCALC_EXIT_DECODING_ERROR`, etc. But that doesn't have to be in this patch.

Reply:
Good point, I added constants in https://github.com/metabrainz/picard/pull/2358/commits/7f6bccdb1521b3eba9135b329fc4bf35fefdd3cd

Reply:
Good point. I added constants in https://github.com/metabrainz/picard/pull/2358/commits/3832719ed8f57086273eb9f971bbec08a763256a